pilot/pkg/model/push_context.go

```diff
@@ -83,6 +83,7 @@ type PushContext struct {
 
 	// ServiceByHostnameAndNamespace has all services, indexed by hostname then namespace.
 	ServiceByHostnameAndNamespace map[host.Name]map[string]*Service `json:"-"`
+	ServiceByHostname             map[host.Name]*Service            `json:"-"`
```

Comment:
What happens if multiple services with the same name exists in different namespaces?

Reply:
The current behavior today is preserved, basically we pick one arbitrarily. Its a gateway do there is no scoping so we can't make a smarter decision like for sidecar

Reply:
note that hostname is the fqdn so this can only happen with service entry

Reply:
Yeah. I am wondering from service entry perspective only. But this  makes sense